packages/node-runtime-worker-thread/src/index.ts

```diff
@@ -222,7 +231,7 @@ class WorkerRuntime implements Runtime {
 
   async interrupt() {
     await this.initWorkerPromise;
-    return this.childProcessRuntime.interrupt();
+    return this.childProcess.kill('SIGINT');
```

Comment:
This means that we're now losing all state (e.g. variables, global object, etc.) in the shell when using Ctrl+C, correct?

That was one of the reasons why we wanted to use a Worker thread, and why we wanted to remove the child process indirection rather than the Worker thread one here ... I remember we talked about making the integration with Electron's web workers possible, I guess that's out of the question here?

Reply:
Cool, yeah – I hadn't tested this, so if everything's working fine, then feel free to disregard me 🙂 Basit can probably confirm once he's back

Reply:
![Screenshot 2024-07-23 at 5 53 35 PM](https://github.com/user-attachments/assets/05d54f6b-488f-4909-a4d4-07100662eca3)
[this pr on the left | current Compass on the right]
Basit mentioned this might not work with ctrl + c on windows, and we may want to use a new worker directly, and use worker.interupt, instead of spawning a process. Looking into that now.